src/background/icon-manager.ts

```diff
@@ -54,10 +64,8 @@ export default class IconManager {
             return;
         }
         IconManager.iconState.active = true;
-        chrome.browserAction.setIcon({
-            path: IconManager.ICON_PATHS.active,
-        });
-        IconManager.handleUpdate();
+
+        this.setTheme(UserStorage.settings.theme.mode);
```

Comment:
Please try avoiding using the global object here. Maybe it's possible to pass the theme as an argument or have a single method like `setIcon({dark: true, active: true});`.